geps/gep-1651.md

```diff
@@ -0,0 +1,408 @@
+
+It is RECOMMENDED that in-cluster gateways SHOULD NOT support 'Private' routability. Kubernetes doesn't have
+a concept of 'Private' routability for Services. In the future this may change upstream.
+
+## Examples
+
+#### 1. Request a GatewayAddress that is routable within the same cluster
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 2. Request a GatewayAddress with a specific routability and address
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+  addresses:
+  - value: 10.0.0.8
+```
+
+#### 3. Request a GatewayAddress that is routable on the public internet
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 4. Request a GatewayAddress that is a cloud provider's VPC
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Private
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+## Alternatives
+
+### Introducing new GatewayAddress Types
+
+We could introduce additional `AddressTypes` (ie. `ClusterLocalIPAddress`) but
+this would lead to a combinatorial explosion as new dimensions (ie. IPv6) are
+introduced.
+
+From: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1451246877
+
+> Although this makes sense in isolation, I'm worried about the long term impacts this could have. In my opinion, ClusterLocal is a modifier, not exactly an address type. For example, it's possible in the future that we'll have a way to provision cluster-local DNS names, we may want to use the same kind of mechanism to request a ClusterLocal DNS name for the Gateway.
+>
+> It's also possible that users will want to explicitly request an IP Families (v4, v6, or both). I'd really hate to get into a situation where we have the following options:
+>
+>   IPAddress
+>   IPv4Address
+>   IPv6Address
+>   ClusterLocalIPAddress
+>   ClusterLocalIPv4Address
+>   ClusterLocalIPv6Address
+>
+> For each dimension we avoid adding a separate field for and instead try to embed into a single name, we risk this kind of name explosion. Of course none of the above even begins to cover my idea of NetworkLocal which could further complicate this.
+
+### Scope/reachability/routability field on GatewayAddress
+
+This would allow Gateways to have multiple scopes.
+
+From: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1486271913
```

Comment:
Please make this a link (https://deploy-preview-1653--kubernetes-sigs-gateway-api.netlify.app/geps/gep-1651/#scopereachabilityroutability-field-on-gatewayaddress)